### PR TITLE
Remove unused parameters and adjust dependent methods.

### PR DESCRIPTION
**Pull request by lorenzgerber**
In the current implementation of ExtractedMatrix, it is not possible to change the size of the matrix/number of scans, hence the numberOfIons, numberOfScans parameters are not used.

### CODE REVIEW

File: chemclipse/plugins/org.eclipse.chemclipse.msd.model/src/org/eclipse/chemclipse/msd/model/matrix/ExtractedMatrix.java [hunk 1]
```diff
@@ -87,62 +87,63 @@ private Boolean checkHighRes( int limit ) {
...
 	}
 
 	public int getNumberOfIons() {
+
 		return this.numberOfIons;
 	}
 
-	public void updateSignal(float[][] signal, int numberOfScans, int nubmerOfIons ) {
-		
+	public void updateSignal(float[][] signal) {
```
Comment (laeubi) [inline chemclipse/plugins/org.eclipse.chemclipse.msd.model/src/org/eclipse/chemclipse/msd/model/matrix/ExtractedMatrix.java]: If you want going further, you can even omit the signal matrix here as it is already available in this.signal. At least you should not rely on the fact that the passed in array has the desired size (never trust the caller!)

File: chemclipse/plugins/org.eclipse.chemclipse.msd.model/src/org/eclipse/chemclipse/msd/model/matrix/ExtractedMatrix.java [hunk 2]
```diff
@@ -87,62 +87,63 @@ private Boolean checkHighRes( int limit ) {
...
+		return (minMaxMz);
 	}
 
 	public float[][] getMatrix() {
+
 		return this.signal;
 	}
 
 	public int getNumberOfScans() {
+
```
Comment (laeubi) [inline chemclipse/plugins/org.eclipse.chemclipse.msd.model/src/org/eclipse/chemclipse/msd/model/matrix/ExtractedMatrix.java]: NumberOfscanns is the length of the matrix "this.signal" and must not be stored as an extra field

File: chemclipse/plugins/org.eclipse.chemclipse.msd.model/src/org/eclipse/chemclipse/msd/model/matrix/ExtractedMatrix.java [hunk 3]
```diff
@@ -87,62 +87,63 @@ private Boolean checkHighRes( int limit ) {
...
 		return this.numberOfScans;
 	}
 
 	public int getNumberOfIons() {
+
```
Comment (laeubi) [inline chemclipse/plugins/org.eclipse.chemclipse.msd.model/src/org/eclipse/chemclipse/msd/model/matrix/ExtractedMatrix.java]: the number of ions is equal to the length of the first matrix element (this.signal[0].length) and thus could be ommited.

File: chemclipse/plugins/org.eclipse.chemclipse.msd.model/src/org/eclipse/chemclipse/msd/model/matrix/ExtractedMatrix.java (hunk 2 above)
Comment (lorenzgerber) [reply chemclipse/plugins/org.eclipse.chemclipse.msd.model/src/org/eclipse/chemclipse/msd/model/matrix/ExtractedMatrix.java]: removed method altogether and documented the dimensions of signal array in javadoc instead

File: chemclipse/plugins/org.eclipse.chemclipse.msd.model/src/org/eclipse/chemclipse/msd/model/matrix/ExtractedMatrix.java (hunk 3 above)
Comment (lorenzgerber) [reply chemclipse/plugins/org.eclipse.chemclipse.msd.model/src/org/eclipse/chemclipse/msd/model/matrix/ExtractedMatrix.java]: removed method altogether and documented the dimensions of signal array in javadoc instead

**Review by laeubi — APPROVED** [review]

**Pull request merged**